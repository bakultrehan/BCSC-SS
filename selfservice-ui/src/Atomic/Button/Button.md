This BC button is amazing, use it responsibly.

## Examples

### Primary button

Primary buttons allow users to carry out an important action on your service, such as Download or Submit.

```jsx
<Button>Primary Btn</Button>
```

Normal primary dark button:

```jsx
<div class='background-colour'>
  <Button dark>Dark Btn</Button>
</div>
```

### Secondary button

Secondary buttons allow users to carry out a supporting action on your service, such as Back or Cancel.

```jsx
<div>
  <Button secondary>secondary Btn</Button>
</div>
```

secondary button with dark and blue background:

```jsx
<div class='background-colour'>
  <Button secondary dark>
    Dark secondary Btn
  </Button>
</div>
```

### Disabled button

Disabled button :

```jsx
<div class='background-colour'>
  <Button disabled>Disabled</Button>
</div>
```

Disabled secondary button :

```jsx
<div class='background-colour'>
  <Button secondary disabled>
    secondary Disabled Btn
  </Button>
</div>
```

With on click action
click

```vue
<template>
  <Button @click="click">Click me</Button>
</template>
<script>
export default {
  methods: {
    click() {
      console.log('you pressed button');
    }
  }
};
</script>
```
