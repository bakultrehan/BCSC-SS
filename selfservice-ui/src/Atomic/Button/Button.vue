/** * Button atomic component * Primary buttons allow users to carry out an
important action on your service, such as Download or Submit. */

<template>
  <v-btn
    class="btn"
    :type="type"
    :name="name"
    v-bind:class="[
      secondary !== false ? 'secondary-btn' : 'primary',
      dark !== false ? 'dark-calss' : 'normal',
      disabled !== false ? 'disabled' : '',
    
    ]"
    :aria-disabled="disabled"
    :disabled="disabled"
    @click="click"
    v-bind="$attrs"
    :color="color !== '' ? color : undefined"
  >
    <!-- color="#fcba19 !important" -->
    <!-- yellowBtn !== false ? 'yellow-btn':'' -->
    <!-- @slot  Button display name-->
    <slot></slot>
  </v-btn>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

/**
 * All variation of button
 * @displayName  Buttons
 */

@Component
export default class Button extends Vue {
  /**
   * The name for the button.
   * @values button, submit
   */
  @Prop({
    default: 'button'
  })
  private name!: string;

  /**
   * The type for the button.
   */
  @Prop({
    default: 'button'
  })
  private type!: string;

  /**
   *  Dark Background  button.
   */
  @Prop({
    default: false
  })
  private dark!: boolean;

  /**
   *  Dark secondary  button. default will be primary
   */
  @Prop({
    default: false
  })
  private secondary!: boolean;

  /**
   *  Dark disabled  button.
   */
  @Prop({
    default: false
  })
  private disabled!: boolean;
  /**
   *  custom color button. default will be primary
   */
  @Prop({
    default: ''
  })
  private color!: string;

  /**
   * on click
   * @description update on click
   *
   */
  private click() {
    this.$emit('click');
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.btn {
  border: none;
  padding: 10px 32px !important;
  min-height: 45px;
  display: block;
  text-align: center;
  text-decoration: none;
  font-size: 18px;
  font-weight: 700;
  letter-spacing: 1px;
  font-family: 'BCSans', 'Noto Sans', Verdana, Arial, sans-serif;
  cursor: pointer;
}

.btn:focus:not(.disabled) {
  outline: 4px solid #3b99fc;
  outline-offset: 1px;
}
.bc-btn:active:not(.disabled) {
  opacity: 1;
}
.secondary-btn {
  background: none !important;
  border: 2px solid $BCgovBlue5;
  color: $BCgovBlue5 !important;
  font-weight: 700;
}
.secondary-btn:hover:hover:not(.disabled) {
  opacity: 0.8;
  text-decoration: underline;
}
.secondary-btn:focus:hover:not(.disabled) {
  outline-offset: 1px;
  outline: 4px solid $BCgovABlue2;
}
.secondary-btn.normal:hover:hover:not(.disabled) {
  background-color: $BCgovBlue5 !important;
  color: $BCgovFontColorInverted !important;
}
.secondary-btn.dark-calss {
  background-color: $BCgovBlue5 !important;
  border: 2px solid $BCgovFontColorInverted;
  color: $BCgovFontColorInverted !important;
}

.secondary-btn.dark-calss:hover:hover:not(.disabled) {
  text-decoration: underline;
  background-color: $BCgovFontColorInverted !important;
  color: $BCgovFontColorInvertedDark !important;
  opacity: 1;
}

.primary {
  border: none;
  // background-color: #003366;
  border-radius: 4px;
  color: white;
  text-decoration: none;
}

.primary.normal {
  background-color: $BCgovBlue5 !important;
  color: white !important;
}
.primary.dark-calss {
  background-color: $BCgovFontColorInverted !important;
  color: $BCgovFontColorInvertedDark !important;
}
.primary.normal:hover:not(.disabled) {
  opacity: 0.8;
}

.dark-calss:hover:not(.disabled) {
  background-color: #f2f2f2;
  text-decoration: underline;
}
.dark-calss:active {
  background-color: $BCgovFontColorInverted !important;
}
.background-colour {
  background-color: $BCgovBlue5;
  height: 100px;
  width: 300px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.disabled {
  cursor: not-allowed;
  opacity: 0.3;
}
.secondary-btn.theme--light.v-btn.v-btn--disabled:not(.v-btn--flat):not(.v-btn--text):not(.v-btn--outlined) {
  background-color: $BCgovFontColorInverted !important;
}
.primary.theme--light.v-btn.v-btn--disabled:not(.v-btn--flat):not(.v-btn--text):not(.v-btn--outlined) {
  background-color: $BCgovBlue5 !important;
}
</style>
