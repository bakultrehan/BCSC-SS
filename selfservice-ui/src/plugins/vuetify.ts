import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
