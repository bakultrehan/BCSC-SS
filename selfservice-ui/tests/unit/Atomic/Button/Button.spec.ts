import { shallowMount, createLocalVue } from '@vue/test-utils';
import Button from '@/Atomic/Button/Button.vue';
import vuetify from 'vuetify';

describe('Button.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);
    wrapper = shallowMount(Button, {
      localVue
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
