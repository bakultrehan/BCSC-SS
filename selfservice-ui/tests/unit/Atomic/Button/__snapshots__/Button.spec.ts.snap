// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders props when passed 1`] = `
<v-btn-stub
  activeclass=""
  class="btn primary normal"
  name="button"
  tag="button"
  type="button"
/>
`;
